worlds/kh2/__init__.py

```diff
@@ -23,119 +33,137 @@ class KingdomHearts2Web(WebWorld):
     )]
 
 
-# noinspection PyUnresolvedReferences
 class KH2World(World):
     """
     Kingdom Hearts II is an action role-playing game developed and published by Square Enix and released in 2005.
     It is the sequel to Kingdom Hearts and Kingdom Hearts: Chain of Memories, and like the two previous games,
     focuses on Sora and his friends' continued battle against the Darkness.
     """
-    game: str = "Kingdom Hearts 2"
+    game = "Kingdom Hearts 2"
     web = KingdomHearts2Web()
-    data_version = 1
-    required_client_version = (0, 4, 0)
-    option_definitions = KH2_Options
-    item_name_to_id = {name: data.code for name, data in item_dictionary_table.items()}
-    location_name_to_id = {item_name: data.code for item_name, data in all_locations.items() if data.code}
+
+    required_client_version = (0, 4, 4)
+    options_dataclass = KingdomHearts2Options
+    options: KingdomHearts2Options
+    item_name_to_id = {item: item_id
+                       for item_id, item in enumerate(item_dictionary_table.keys(), 0x130000)}
+    location_name_to_id = {item: location
+                           for location, item in enumerate(all_locations.keys(), 0x130000)}
     item_name_groups = item_groups
+    # list of KH2Item
+    keyblade_ability_pool = list()
+    goofy_get_bonus_abilities = list()
+    goofy_weapon_abilities = list()
+    donald_get_bonus_abilities = list()
+    donald_weapon_abilities = list()
+
+    slot_data_goofy_weapon = dict()
+    slot_data_sora_weapon = dict()
+    slot_data_donald_weapon = dict()
+    visitlocking_dict: Dict[str, int]
+    plando_locations: Dict[str, str]
+    lucky_emblem_amount: int
+    lucky_emblem_required: int
+    bounties_required: int
+    bounties_amount: int
+    filler_items: List[str]
+    item_quantity_dict: Dict[str, int]
```

Comment:
You probably want `Counter`s for these? I'm not entirely sure if you have anything that would make those easier to use. Do note they're a bit more expensive than just standard dicts so if they don't help probably don't.

Reply:
yeah I dont think it would be worth